### ISSUE TITLE
[1.4.2 beta 9] - Cordova app crashes/keeps trying to download a new version from server

### ISSUE THREAD

**Issue by adamgins** [meteor/meteor#7876]
using Meteor  1.4.2-beta 9 w/WiredTiger 3.2 on Compose.io
cordova:cordova-plugin-wkwebview-engine@1.1.0 as per https://github.com/meteor/meteor/issues/7018#issuecomment-251074746

Load of meteor packages, here's a subset that may be related:

```
accounts-base@1.2.12-beta.9
accounts-facebook@1.0.11-beta.9
accounts-google@1.0.11-beta.9
accounts-oauth@1.1.14-beta.9
accounts-password@1.3.1-beta.9
accounts-twitter@1.1.12-beta.9
adamgins:microsoft-oauth@0.0.2
alisalaah:hogan@1.0.1
allow-deny@1.0.5
autoupdate@1.3.11
babel-compiler@6.9.2-beta.9
babel-runtime@0.1.12-beta.9
base64@1.0.10-beta.9
binary-heap@1.0.10-beta.9
blaze@2.1.9
blaze-html-templates@1.0.5
blaze-tools@1.0.10
boilerplate-generator@1.0.11-beta.9
bozhao:accounts-instagram@0.2.2
browser-policy@1.0.9
browser-policy-common@1.0.10
browser-policy-content@1.0.11
browser-policy-framing@1.0.11
buzzy-buzz:avatar@0.9.2
buzzy-buzz:common-globals@0.1.3
buzzy-buzz:resources-core@0.1.5
caching-compiler@1.1.8-beta.9
caching-html-compiler@1.0.7
callback-hook@1.0.10-beta.9
check@1.2.4-beta.9
copleykj:jquery-autosize@1.17.8
crosswalk@1.7.0-beta.9
dburles:google-maps@1.1.5
ddp@1.2.5
ddp-client@1.3.1
ddp-common@1.2.6
ddp-rate-limiter@1.0.5
ddp-server@1.3.10
deps@1.0.12
diff-sequence@1.0.7-beta.9
dispatch:deep-link@4.0.4
djedi:sanitize-html@1.11.3
ecmascript@0.5.9-beta.9
ecmascript-runtime@0.3.15-beta.9
ejson@1.0.13-beta.9
email@1.1.18-beta.9
facebook@1.2.10-beta.9
fastclick@1.0.13-beta.9
force-ssl@1.0.13-beta.9
fortawesome:fontawesome@4.6.3
froala:editor@2.3.4
froala:editor-reactive@2.3.4
froatsnook:request@2.74.0
geojson-utils@1.0.9
google@1.1.15-beta.9
handlebars@1.0.7
harrison:papa-parse@1.1.1
hot-code-push@1.0.4
html-tools@1.0.11
htmljs@1.0.11
http@1.2.10-beta.9
huttonr:bootstrap3@3.3.6_13
huttonr:bootstrap3-assets@3.3.6_3
id-map@1.0.9-beta.9
iron:controller@1.0.12
iron:core@1.0.11
iron:dynamic-template@1.0.12
iron:layout@1.0.12
iron:location@1.0.11
iron:middleware-stack@1.1.0
iron:router@1.0.13
iron:url@1.0.11
jeremy:selectize@0.12.1_5
johnantoni:meteor-appear@0.0.2
jparker:crypto-core@0.1.0
jparker:crypto-hmac@0.1.0
jparker:crypto-md5@0.1.1
jparker:crypto-sha256@0.1.1
jparker:gravatar@0.5.1
jquery@1.11.10-beta.9
konecty:autolinker@1.0.3
launch-screen@1.0.12
less@2.7.6-beta.9
linto:jquery-ui@1.11.2
livedata@1.0.18
localstorage@1.0.12-beta.9
logging@1.1.16-beta.9
lookback:emails@0.7.5
lukemadera:social-share@1.2.0
manuelschoebel:cheerio@0.0.1
maxharris9:connection-banner@0.5.0
mdg:geolocation@1.3.0
mdg:reload-on-resume@1.0.4
meteor@1.2.18-beta.9
meteor-base@1.0.4
meteorhacks:aggregate@1.3.0
meteorhacks:async@1.0.0
meteorhacks:collection-utils@1.2.0
meteorhacks:inject-initial@1.0.4
meteorhacks:kadira@2.30.0
meteorhacks:meteorx@1.4.1
meteorhacks:npm@1.5.0
meteorhacks:picker@1.0.3
meteorhacks:ssr@2.2.0
meteorhacks:subs-manager@1.6.4
meteorhacks:unblock@1.1.0
minifier-css@1.2.14
minifier-js@1.2.15-beta.9
minimongo@1.1.0-ddp-batching-beta.0
mizzao:timesync@0.4.0
mobile-experience@1.0.4
mobile-status-bar@1.0.12
modules@0.7.7-beta.9
modules-runtime@0.7.7-beta.9
momentjs:moment@2.15.1
mongo@1.1.13-beta.9
mongo-id@1.0.6-beta.9
mongo-livedata@1.0.12
mquandalle:ismobile@0.1.0
mrt:q@1.0.1
mrt:twilio-meteor@1.1.0
mrt:twit@0.2.0
natestrauser:filepicker-plus@2.0.0
npm-bcrypt@0.9.1
npm-container@1.2.0
npm-mongo@1.5.50
oauth@1.1.12-beta.9
oauth1@1.1.10
oauth2@1.1.11-beta.9
observe-sequence@1.0.12
okgrow:analytics@1.0.9
okgrow:router-autoscroll@0.1.8
ordered-dict@1.0.8
peerlibrary:aws-sdk@2.4.9_1
peerlibrary:blocking@0.5.2
peppelg:bootstrap-3-modal@1.0.4
percolate:google-api@1.0.3
pfafman:filesaver@1.3.2
promise@0.8.8-beta.9
raix:eventemitter@0.1.3
raix:eventstate@0.0.4
raix:handlebar-helpers@0.2.5
raix:push@3.1.0-pre.1
raix:ui-dropped-event@0.0.7
random@1.0.10
rate-limit@1.0.5
reactive-dict@1.1.8
reactive-var@1.0.11-beta.9
reload@1.1.11-beta.9
retry@1.0.8
reywood:publish-composite@1.4.2
richsilv:owl-carousel@0.1.0
risul:moment-timezone@0.5.0_5
routepolicy@1.0.12-beta.9
sacha:juice@0.1.4
sacha:spin@2.3.1
service-configuration@1.0.11-beta.9
session@1.1.7-beta.9
sha@1.0.8
shell-server@0.2.1
simple:reactive-method@1.0.2
spacebars@1.0.13
spacebars-compiler@1.0.13
srp@1.0.9
standard-minifier-css@1.2.2-beta.9
standard-minifier-js@1.2.1-beta.9
templating@1.2.15
templating-compiler@1.2.15
templating-runtime@1.2.15
templating-tools@1.0.5
tracker@1.1.1-beta.9
tsega:bootstrap3-datetimepicker@4.17.37_1
twitter@1.1.13-release-testing.0
ui@1.0.12
underscore@1.0.10-beta.9
url@1.0.11-beta.9
webapp@1.3.12-beta.9
webapp-hashing@1.0.9

```

for Cordova specifically:

```
cordova-custom-config@2.0.3
cordova-plugin-calendar@4.4.7
cordova-plugin-contacts@2.2.0
cordova-plugin-device@1.1.3
cordova-plugin-file@4.3.0
cordova-plugin-geolocation@2.4.0
cordova-plugin-meteor-webapp@https://github.com/meteor/cordova-plugin-meteor-webapp.git#8bf95eed3f313299fc2de33658866278eea2cdc5
cordova-plugin-splashscreen@4.0.0
cordova-plugin-statusbar@2.2.0
cordova-plugin-wkwebview-engine@1.1.0
```

I am getting some errors when I run it: `meteor run ios-device --settings settings.json --mobile-server https://<public server address>`

I deploy my application to a Galaxy server available to the internet. 

Via XCode 8, I delply the Cordova app to me physical iOS10 phone.

If I leave the app open it crashes/times out and trys to reload a new version from the server, here are the logs:

```
2016-10-06 21:30:34.461077 Buzzy[4942:1113060] DiskCookieStorage changing policy from 2 to 0, cookie file: file:///private/var/mobile/Containers/Data/Application/65D1B431-2B43-4910-97C0-9F09689E452E/Library/Cookies/Cookies.binarycookies
2016-10-06 21:30:34.510399 Buzzy[4942:1113060] Apache Cordova native platform version 4.2.1 is starting.
2016-10-06 21:30:34.511003 Buzzy[4942:1113060] Multi-tasking -> Device: YES, App: YES
2016-10-06 21:30:34.519564 Buzzy[4942:1113060] 

Started backup to iCloud! Please be careful.
Your application might be rejected by Apple if you store too much data.
For more information please read "iOS Data Storage Guidelines" at:
https://developer.apple.com/icloud/documentation/data-storage/
To disable web storage backup to iCloud, set the BackupWebStorage preference to "local" in the Cordova config.xml file


2016-10-06 21:30:34.591678 Buzzy[4942:1113060] CDVWKWebViewEngine will reload WKWebView if required on resume
2016-10-06 21:30:34.591729 Buzzy[4942:1113060] Using WKWebView
2016-10-06 21:30:34.592135 Buzzy[4942:1113060] [CDVTimer][handleopenurl] 0.064015ms
2016-10-06 21:30:34.593781 Buzzy[4942:1113060] Unlimited access to network resources
2016-10-06 21:30:34.593873 Buzzy[4942:1113060] [CDVTimer][intentandnavigationfilter] 1.695991ms
2016-10-06 21:30:34.594038 Buzzy[4942:1113060] [CDVTimer][gesturehandler] 0.092983ms
2016-10-06 21:30:34.654138 Buzzy[4942:1113060] Serving asset bundle version: 0d3a1407b1794915b2ec3a071525890ec58c847c
[INFO] GCDWebServer started on port 12456 and reachable at http://localhost:12456/
2016-10-06 21:30:34.656736 Buzzy[4942:1113060] [CDVTimer][webapplocalserver] 62.660992ms
2016-10-06 21:30:34.666776 Buzzy[4942:1113060] [CDVTimer][statusbar] 9.990036ms
2016-10-06 21:30:34.683270 Buzzy[4942:1113060] [CDVTimer][splashscreen] 16.433001ms
2016-10-06 21:30:34.684259 Buzzy[4942:1113060] [CDVTimer][file] 0.941992ms
2016-10-06 21:30:34.684290 Buzzy[4942:1113060] [CDVTimer][TotalPluginStartup] 92.288017ms
2016-10-06 21:30:35.396436 Buzzy[4942:1113060] createNotificationChecker
2016-10-06 21:30:35.396534 Buzzy[4942:1113060] not coldstart
2016-10-06 21:30:35.398687 Buzzy[4942:1113060] active
2016-10-06 21:30:35.399019 Buzzy[4942:1113060] PushPlugin skip clear badge
2016-10-06 21:30:37.343810 Buzzy[4942:1113352] Push Plugin register called
2016-10-06 21:30:37.343871 Buzzy[4942:1113352] PushPlugin.register: setting badge to false
2016-10-06 21:30:37.343887 Buzzy[4942:1113352] PushPlugin.register: clear badge is set to 0
2016-10-06 21:30:37.343897 Buzzy[4942:1113352] PushPlugin.register: better button setup
2016-10-06 21:30:37.346147 Buzzy[4942:1113352] GCM Sender ID (null)
2016-10-06 21:30:37.346181 Buzzy[4942:1113352] Using APNS Notification
2016-10-06 21:30:38.031678 Buzzy[4942:1113060] Push Plugin register success: <c82e1e28 1d1136d5 d1a650ae a36ba817 a3bac30c a12c3ebf f9b30bb4 c716326a>
2016-10-06 21:30:38.952244 Buzzy[4942:1113060] Start downloading asset manifest from: manifest.json -- https://a.buzzy.buzz/__cordova/
2016-10-06 21:30:39.960891 Buzzy[4942:1113306] Downloaded asset manifest for version: 369eea732411e1e15b50da00173af55ac4a133ec
2016-10-06 21:30:39.973010 Buzzy[4942:1113306] Start downloading assets from bundle with version: 369eea732411e1e15b50da00173af55ac4a133ec
2016-10-06 21:30:42.514644 Buzzy[4942:1113060] active
2016-10-06 21:30:42.514844 Buzzy[4942:1113060] PushPlugin skip clear badge
2016-10-06 21:30:42.557794 Buzzy[4942:1113060] Push Plugin register success: <c82e1e28 1d1136d5 d1a650ae a36ba817 a3bac30c a12c3ebf f9b30bb4 c716326a>
2016-10-06 21:30:45.108292 Buzzy[4942:1113060] THREAD WARNING: ['InAppBrowser'] took '119.844238' ms. Plugin should use a background thread.
2016-10-06 21:30:45.120402 Buzzy[4942:1113060] THREAD WARNING: ['InAppBrowser'] took '11.629883' ms. Plugin should use a background thread.
2016-10-06 21:30:47.108524 Buzzy[4942:1113314] WF: === Starting WebFilter logging for process Buzzy
2016-10-06 21:30:47.108750 Buzzy[4942:1113314] WF: _userSettingsForUser mobile: {
    filterBlacklist =     (
    );
    filterWhitelist =     (
    );
    restrictWeb = 1;
    useContentFilter = 0;
    useContentFilterOverrides = 0;
    whitelistEnabled = 0;
}
2016-10-06 21:30:47.109174 Buzzy[4942:1113314] WF: _WebFilterIsActive returning: NO
2016-10-06 21:30:48.586106 Buzzy[4942:1113060] [MC] System group container for systemgroup.com.apple.configurationprofiles path is /private/var/containers/Shared/SystemGroup/systemgroup.com.apple.configurationprofiles
2016-10-06 21:30:48.588990 Buzzy[4942:1113060] [MC] Reading from public effective user settings.
2016-10-06 21:30:54.434397 Buzzy[4942:1113352] [CDVTimer][assetBundleDownload] 14461.089015ms
2016-10-06 21:30:54.436559 Buzzy[4942:1113304] Finished downloading new asset bundle version: 369eea732411e1e15b50da00173af55ac4a133ec
2016-10-06 21:30:56.592935 Buzzy[4942:1113314] WF: _userSettingsForUser mobile: {
    filterBlacklist =     (
    );
    filterWhitelist =     (
    );
    restrictWeb = 1;
    useContentFilter = 0;
    useContentFilterOverrides = 0;
    whitelistEnabled = 0;
}
2016-10-06 21:30:56.593244 Buzzy[4942:1113314] WF: _WebFilterIsActive returning: NO
2016-10-06 21:31:02.545244 Buzzy[4942:1113304] 0x170341ef0 Copy matching assets reply: XPC_TYPE_DICTIONARY  <dictionary: 0x170341ef0> { count = 1, transaction: 0, voucher = 0x0, contents =
    "Result" => <int64: 0x170423a80>: 29
}
2016-10-06 21:31:02.546963 Buzzy[4942:1113304] 0x174340a50 Daemon configuration query reply: XPC_TYPE_DICTIONARY  <dictionary: 0x174340a50> { count = 2, transaction: 0, voucher = 0x0, contents =
    "Dictionary" => <dictionary: 0x1743413f0> { count = 1, transaction: 0, voucher = 0x0, contents =
        "ServerURL" => <dictionary: 0x174341290> { count = 3, transaction: 0, voucher = 0x0, contents =
            "com.apple.CFURL.magic" => <uuid: 0x174441890> C3853DCC-9776-4114-B6C1-FD9F51944A6D
            "com.apple.CFURL.string" => <string: 0x1744478f0> { length = 30, contents = "https://mesu.apple.com/assets/" }
            "com.apple.CFURL.base" => <null: 0x1b1f03e80>: null-object
        }
    }
    "Result" => <int64: 0x17422b520>: 0
}
2016-10-06 21:31:02.547093 Buzzy[4942:1113304] [MobileAssetError:29] Unable to copy asset information from https://mesu.apple.com/assets/ for asset type com.apple.MobileAsset.TextInput.SpellChecker
2016-10-06 21:31:03.791428 Buzzy[4942:1113314] WF: _userSettingsForUser mobile: {
    filterBlacklist =     (
    );
    filterWhitelist =     (
    );
    restrictWeb = 1;
    useContentFilter = 0;
    useContentFilterOverrides = 0;
    whitelistEnabled = 0;
}
2016-10-06 21:31:03.791568 Buzzy[4942:1113314] WF: _WebFilterIsActive returning: NO
2016-10-06 21:31:10.678349 Buzzy[4942:1113314] WF: _userSettingsForUser mobile: {
    filterBlacklist =     (
    );
    filterWhitelist =     (
    );
    restrictWeb = 1;
    useContentFilter = 0;
    useContentFilterOverrides = 0;
    whitelistEnabled = 0;
}
2016-10-06 21:31:10.679820 Buzzy[4942:1113314] WF: _WebFilterIsActive returning: NO
2016-10-06 21:31:12.888383 Buzzy[4942:1113060] IAB.close() called but it was already closed.
2016-10-06 21:32:35.585828 Buzzy[4942:1113845] App startup timed out, reverting to last known good version
2016-10-06 21:32:35.607559 Buzzy[4942:1113060] Serving asset bundle version: 369eea732411e1e15b50da00173af55ac4a133ec
2016-10-06 21:32:35.607770 Buzzy[4942:1113060] IAB.close() called but it was already closed.
2016-10-06 21:32:36.720706 Buzzy[4942:1113934] Push Plugin register called
2016-10-06 21:32:36.720772 Buzzy[4942:1113934] PushPlugin.register: setting badge to false
2016-10-06 21:32:36.720793 Buzzy[4942:1113934] PushPlugin.register: clear badge is set to 0
2016-10-06 21:32:36.720803 Buzzy[4942:1113934] PushPlugin.register: better button setup
2016-10-06 21:32:36.721026 Buzzy[4942:1113934] GCM Sender ID (null)
2016-10-06 21:32:36.721109 Buzzy[4942:1113934] Using APNS Notification
2016-10-06 21:32:36.736731 Buzzy[4942:1113060] Push Plugin register success: <c82e1e28 1d1136d5 d1a650ae a36ba817 a3bac30c a12c3ebf f9b30bb4 c716326a>
2016-10-06 21:34:35.609232 Buzzy[4942:1114615] App startup timed out, reverting to last known good version
2016-10-06 21:34:35.627874 Buzzy[4942:1113060] Serving asset bundle version: 0d3a1407b1794915b2ec3a071525890ec58c847c
2016-10-06 21:34:35.628066 Buzzy[4942:1113060] IAB.close() called but it was already closed.
2016-10-06 21:34:37.327920 Buzzy[4942:1114631] Push Plugin register called
2016-10-06 21:34:37.327983 Buzzy[4942:1114631] PushPlugin.register: setting badge to false
2016-10-06 21:34:37.328000 Buzzy[4942:1114631] PushPlugin.register: clear badge is set to 0
2016-10-06 21:34:37.328010 Buzzy[4942:1114631] PushPlugin.register: better button setup
2016-10-06 21:34:37.328070 Buzzy[4942:1114631] GCM Sender ID (null)
2016-10-06 21:34:37.328091 Buzzy[4942:1114631] Using APNS Notification
2016-10-06 21:34:37.334630 Buzzy[4942:1113060] Push Plugin register success: <c82e1e28 1d1136d5 d1a650ae a36ba817 a3bac30c a12c3ebf f9b30bb4 c716326a>
2016-10-06 21:34:40.128898 Buzzy[4942:1113060] Start downloading asset manifest from: manifest.json -- https://a.buzzy.buzz/__cordova/
2016-10-06 21:34:42.951863 Buzzy[4942:1114628] Downloaded asset manifest for version: 369eea732411e1e15b50da00173af55ac4a133ec
2016-10-06 21:34:42.953234 Buzzy[4942:1114628] Download failure: Skipping downloading blacklisted version
2016-10-06 21:36:35.629542 Buzzy[4942:1115043] App startup timed out, reverting to last known good version
```

see 
`2016-10-06 21:34:42.953234 Buzzy[4942:1114628] Download failure: Skipping downloading blacklisted version
2016-10-06 21:36:35.629542 Buzzy[4942:1115043] App startup timed out, reverting to last known good version`

The screen flashes white, then the app shows again. 

So a couple of things:

1) why is the app trying to reload from the server when the installed the app from the same local version via Xcode as I loaded on the server?

2) The app seems to shut down and restart / reload from time to time... but not sure if this is just a timeout?

Ah one more things I see the following when I start my app up on my local machine:

```
%% "cordova-plugin-compat" is required by (cordova-plugin-file) and cannot be removed (hint: use -f or --force)
cordova-custom-config: Skipping auto-restore of config file backup(s)
cordova-custom-config: Installing plugin dependencies...==========      ] 75% 14.4s
cordova-custom-config: Installing plugin dependencies...==========      ] 75% 15.0s
cordova-custom-config: Installing plugin dependencies...==========      ] 75% 15.6s
cordova-custom-config: Skipping auto-restore of config file backup(s)===] 100% 0.0s
cordova-custom-config: Applied custom config from config.xml to /Users/adamginsburg/Documents/development/buzzy7/.meteor/local/cordova-build/platforms/ios/Buzzy/Buzzy-Info.plist

```

**Comment by benjamn** [comment]
@martijnwalraven I included your Cordova updates in `1.4.2-beta.9`, so perhaps you have an intuition what might be wrong here?

**Comment by martijnwalraven** [comment]
@adamgins: One thing I notice is that your app includes `cordova-plugin-meteor-webapp@https://github.com/meteor/cordova-plugin-meteor-webapp.git#8bf95eed3f313299fc2de33658866278eea2cdc5`, which you probably added manually. Try to remove it so the new 1.4.1 version of the plugin will be used, which is the default in Meteor 1.4.2-beta.9.

Not sure if this affects your issue, but can you try again with the new version just in case?

App startup timed out usually signals that there is some kind of error in your JavaScript that prevents the app from starting up normally. That may also happen because of an error with a Cordova plugin. 

`1.4.2-beta.9` updates Cordova and the iOS and Android platforms to their latest versions, which could lead to incompatibilities with non-core plugins.

**Comment by adamgins** [comment]
@martijnwalraven @benjamn  thanks... ok in process of deploying now and will test that out. 

One of the other things is I see this error when I initially run it on my dev machine... then I run it again and it goes away:
```
%% Failed to install 'cordova-plugin-inappbrowser':CordovaError: Cannot find plugin.xml for plugin "cordova-plugin-inappbrowser". Please try adding it again.
  at new PluginInfo (/Users/adamginsburg/.meteor/packages/meteor-tool/.1.4.2-beta.9.1l7pzvr++os.osx.x86_64+web.browser+web.cordova/mt-os.osx.x86_64/dev_bundle/lib/node_modules/cordova-common/src/PluginInfo/PluginInfo.js:351:15)
  at copyPlugin (/Users/adamginsburg/.meteor/packages/meteor-tool/.1.4.2-beta.9.1l7pzvr++os.osx.x86_64+web.browser+web.cordova/mt-os.osx.x86_64/dev_bundle/lib/node_modules/cordova-lib/src/plugman/install.js:669:22)
  at /Users/adamginsburg/.meteor/packages/meteor-tool/.1.4.2-beta.9.1l7pzvr++os.osx.x86_64+web.browser+web.cordova/mt-os.osx.x86_64/dev_bundle/lib/node_modules/cordova-lib/src/plugman/install.js:381:17
  at _fulfilled (/Users/adamginsburg/.meteor/packages/meteor-tool/.1.4.2-beta.9.1l7pzvr++os.osx.x86_64+web.browser+web.cordova/mt-os.osx.x86_64/dev_bundle/lib/node_modules/q/q.js:787:54)
  at self.promiseDispatch.done (/Users/adamginsburg/.meteor/packages/meteor-tool/.1.4.2-beta.9.1l7pzvr++os.osx.x86_64+web.browser+web.cordova/mt-os.osx.x86_64/dev_bundle/lib/node_modules/q/q.js:816:30)
  at Promise.promise.promiseDispatch (/Users/adamginsburg/.meteor/packages/meteor-tool/.1.4.2-beta.9.1l7pzvr++os.osx.x86_64+web.browser+web.cordova/mt-os.osx.x86_64/dev_bundle/lib/node_modules/q/q.js:749:13)
  at /Users/adamginsburg/.meteor/packages/meteor-tool/.1.4.2-beta.9.1l7pzvr++os.osx.x86_64+web.browser+web.cordova/mt-os.osx.x86_64/dev_bundle/lib/node_modules/q/q.js:557:44
  at flush (/Users/adamginsburg/.meteor/packages/meteor-tool/.1.4.2-beta.9.1l7pzvr++os.osx.x86_64+web.browser+web.cordova/mt-os.osx.x86_64/dev_bundle/lib/node_modules/q/q.js:108:17)
  at nextTickCallbackWith0Args (node.js:420:9)
  at process._tickCallback (node.js:349:13)

=> Errors executing Cordova commands:                                              
                                                                                   
   While adding plugin cordova-plugin-inappbrowser@1.5.0 to Cordova project:       
   Cordova error: Cannot find plugin.xml for plugin "cordova-plugin-inappbrowser". Please try adding it again.
   (If the error message contains suggestions for a fix, note that this may not apply to the Meteor integration. You can try running again
   with the --verbose option to help diagnose the issue.)
                                                                                   
/Users/adamginsburg/.meteor/packages/meteor-tool/.1.4.2-beta.9.1l7pzvr++os.osx.x86_64+web.browser+web.cordova/mt-os.osx.x86_64/isopackets/cordova-support/npm/node_modules/meteor/promise/node_modules/meteor-promise/promise_server.js:177
      throw error;
      ^
ExitWithCode:1
Adams-MacBook-Pro:buzzy7 adamginsburg$ 

```

**Comment by adamgins** [comment]
@martijnwalraven FYI - it seems to be behaving better after removing that plugin (letting the build handle it). I just deployed via Testflight and seems to be running OK... harder to debug Testflight app, as I can't see the console on the Testflight installed app.

**Comment by adamgins** [comment]
@martijnwalraven I got the following issue, when leaving the app open for a few minutes:

The page the app was on started showing a spinner. The app was still there (ie not the hote code push whitescreen)... then the page finally showed up. It seems like it got the `startup timed out` below then tried refreshing.  Here's what I saw on the XCode log:

```
2016-10-07 13:20:12.133322 Buzzy[5855:1309293] App startup timed out, reverting to last known good version
2016-10-07 13:20:12.184105 Buzzy[5855:1308736] Serving asset bundle version: cd6c69da1fc352452c178731b3753d21cdc0bd86
2016-10-07 13:20:14.367470 Buzzy[5855:1309496] Push Plugin register called
2016-10-07 13:20:14.367534 Buzzy[5855:1309496] PushPlugin.register: setting badge to false
2016-10-07 13:20:14.367552 Buzzy[5855:1309496] PushPlugin.register: clear badge is set to 0
2016-10-07 13:20:14.367563 Buzzy[5855:1309496] PushPlugin.register: better button setup
2016-10-07 13:20:14.367809 Buzzy[5855:1309496] GCM Sender ID (null)
2016-10-07 13:20:14.367822 Buzzy[5855:1309496] Using APNS Notification
2016-10-07 13:20:14.378541 Buzzy[5855:1308736] Push Plugin register success: <3b92873a 96ed61af 81a9338a a9cf1a16 6aa4b3c7 cb06a4d5 99ccd51f 4c50fd95>
2016-10-07 13:20:16.524350 Buzzy[5855:1308736] Start downloading asset manifest from: manifest.json -- https://a.buzzy.buzz/__cordova/
2016-10-07 13:20:18.225445 Buzzy[5855:1309293] Downloaded asset manifest for version: 5ffe1e9fcc3f53caecbbfaa510f98e4a29f3a765
2016-10-07 13:20:18.226901 Buzzy[5855:1309293] Download failure: Skipping downloading blacklisted version
```
From Safari console
![screen shot 2016-10-07 at 1 26 32 pm](https://cloud.githubusercontent.com/assets/7233741/19177113/d5ea2cda-8c91-11e6-9ede-4c45da1c0851.png)

**Comment by adamgins** [comment]
@martijnwalraven @abernix any ideas on this critical error in my Corodva app whet crashes with

```
2016-11-13 14:02:32.153712 Buzzy[4657:1636512] [MobileAssetError:29] Unable to copy asset information from https://mesu.apple.com/assets/ for asset type com.apple.MobileAsset.TextInput.SpellChecker
```
I then type something in a text input field... after sometime the app refreshes (obviously the user loses what they were typing)

```

2016-11-13 14:03:53.769885 Buzzy[4657:1636607] App startup timed out, reverting to last known good version
2016-11-13 14:03:53.789554 Buzzy[4657:1636283] Serving asset bundle version: 67d1c6554010cda90cacf6ebefb97315377b52cc
2016-11-13 14:03:55.645748 Buzzy[4657:1637042] Push Plugin register called
2016-11-13 14:03:55.645820 Buzzy[4657:1637042] PushPlugin.register: setting badge to false
2016-11-13 14:03:55.645849 Buzzy[4657:1637042] PushPlugin.register: clear badge is set to 0
2016-11-13 14:03:55.645869 Buzzy[4657:1637042] PushPlugin.register: better button setup
2016-11-13 14:03:55.645984 Buzzy[4657:1637042] GCM Sender ID (null)
2016-11-13 14:03:55.646008 Buzzy[4657:1637042] Using APNS Notification
2016-11-13 14:03:55.651882 Buzzy[4657:1636283] Push Plugin register success: <0d25f167 838005a8 eb1ccede df656199 a09f2400 42c890d9 07cf30c5 ade0288c>
2016-11-13 14:03:58.265869 Buzzy[4657:1636283] Start downloading asset manifest from: manifest.json -- https://buzzy.buzz/__cordova/
2016-11-13 14:03:59.356843 Buzzy[4657:1636996] Downloaded asset manifest for version: 06ea28124eb8221400a40960079115860933896c
2016-11-13 14:03:59.358506 Buzzy[4657:1636996] Download failure: Skipping downloading blacklisted version
2016-11-13 14:04:08.458134 Buzzy[4657:1636283] THREAD WARNING: ['Geolocation'] took '11.032959' ms. Plugin should use a background thread.

```

Any ideas on what's causing this refresh pls? Is it trying to get a new version of the app from the server or spellcheker crashing it?

**Comment by martijnwalraven** [comment]
@adamgins: This seems like an iOS issue, unrelated to Meteor. I'm getting a few hits when I Google for `"Unable to copy asset information from" "com.apple.MobileAsset.TextInput.SpellChecker"`, but no clear diagnosis or solution.

**Comment by adamgins** [comment]
@martijnwalraven thanks. I was thinking that others must have hit this.. pretty much any app that has a text field on iOS/Cordova.

Do you know if any Cordova settings params to disable it?

I did see this post http://stackoverflow.com/questions/39868842/error-in-ios-10-unable-to-copy-asset-information-from-https-mesu-apple-com-a but I could not find that Edit Scheme option in XCode and not sure if I should be using mobile-config.js to try and control it?

**Comment by adamgins** [comment]
So just to confirm, I think this issue is to do with hot code push and not the spellchecker issue, although that's still a separate issue.

Here's a detail of the reloads/failed reloads - each which seem associated with a screen refresh on Cordova (that would cause users to lose unsubmitted data).

@martijnwalraven seems related to this https://github.com/meteor/meteor/issues/6847 I am going to try and `rm -rf .meteor/local/cordova-build` and see if that helps

BTW I have these timeout settings:

```
App.setPreference("LoadUrlTimeoutValue", 120000);
App.setPreference("WebAppStartupTimeout", 120000);
```

current logs... see reload /skipping issue ...

```
2016-11-13 21:00:02.542506 Buzzy[4988:1717079] Apache Cordova native platform version 4.2.1 is starting.
2016-11-13 21:00:02.543099 Buzzy[4988:1717079] Multi-tasking -> Device: YES, App: YES
2016-11-13 21:00:02.549471 Buzzy[4988:1717079] 

Started backup to iCloud! Please be careful.
Your application might be rejected by Apple if you store too much data.
For more information please read "iOS Data Storage Guidelines" at:
https://developer.apple.com/icloud/documentation/data-storage/
To disable web storage backup to iCloud, set the BackupWebStorage preference to "local" in the Cordova config.xml file


2016-11-13 21:00:02.616660 Buzzy[4988:1717079] CDVWKWebViewEngine will reload WKWebView if required on resume
2016-11-13 21:00:02.616717 Buzzy[4988:1717079] Using WKWebView
2016-11-13 21:00:02.616960 Buzzy[4988:1717079] [CDVTimer][handleopenurl] 0.050008ms
2016-11-13 21:00:02.618548 Buzzy[4988:1717079] Unlimited access to network resources
2016-11-13 21:00:02.618628 Buzzy[4988:1717079] [CDVTimer][intentandnavigationfilter] 1.567006ms
2016-11-13 21:00:02.618831 Buzzy[4988:1717079] [CDVTimer][gesturehandler] 0.154018ms
2016-11-13 21:00:02.689982 Buzzy[4988:1717079] Serving asset bundle version: 258c426bb6147c4fe43935834585825c1c5ef424
[INFO] GCDWebServer started on port 12456 and reachable at http://localhost:12456/
2016-11-13 21:00:02.691808 Buzzy[4988:1717079] [CDVTimer][webapplocalserver] 72.902977ms
2016-11-13 21:00:02.702817 Buzzy[4988:1717079] [CDVTimer][statusbar] 10.908961ms
2016-11-13 21:00:02.718523 Buzzy[4988:1717079] [CDVTimer][splashscreen] 15.619993ms
2016-11-13 21:00:02.719574 Buzzy[4988:1717079] [CDVTimer][file] 0.980973ms
2016-11-13 21:00:02.719626 Buzzy[4988:1717079] [CDVTimer][TotalPluginStartup] 102.742016ms
2016-11-13 21:00:02.724335 Buzzy[4988:1717079] createNotificationChecker
2016-11-13 21:00:02.724382 Buzzy[4988:1717079] not coldstart
2016-11-13 21:00:02.725239 Buzzy[4988:1717079] active
2016-11-13 21:00:02.725371 Buzzy[4988:1717079] PushPlugin skip clear badge
2016-11-13 21:00:04.369443 Buzzy[4988:1717247] Push Plugin register called
2016-11-13 21:00:04.369502 Buzzy[4988:1717247] PushPlugin.register: setting badge to false
2016-11-13 21:00:04.369558 Buzzy[4988:1717247] PushPlugin.register: clear badge is set to 0
2016-11-13 21:00:04.369579 Buzzy[4988:1717247] PushPlugin.register: better button setup
2016-11-13 21:00:04.371698 Buzzy[4988:1717247] GCM Sender ID (null)
2016-11-13 21:00:04.371735 Buzzy[4988:1717247] Using APNS Notification
2016-11-13 21:00:04.378640 Buzzy[4988:1717079] Push Plugin register success: <0d25f167 838005a8 eb1ccede df656199 a09f2400 42c890d9 07cf30c5 ade0288c>
2016-11-13 21:00:07.532335 Buzzy[4988:1717079] Start downloading asset manifest from: manifest.json -- https://buzzy.buzz/__cordova/
2016-11-13 21:00:08.815026 Buzzy[4988:1717251] Downloaded asset manifest for version: 06ea28124eb8221400a40960079115860933896c
2016-11-13 21:00:08.827166 Buzzy[4988:1717251] Start downloading assets from bundle with version: 06ea28124eb8221400a40960079115860933896c
2016-11-13 21:00:15.350467 Buzzy[4988:1717247] [CDVTimer][assetBundleDownload] 6521.530986ms
2016-11-13 21:00:15.353954 Buzzy[4988:1717405] Finished downloading new asset bundle version: 06ea28124eb8221400a40960079115860933896c
2016-11-13 21:00:25.973556 Buzzy[4988:1717079] THREAD WARNING: ['Geolocation'] took '16.937012' ms. Plugin should use a background thread.
2016-11-13 21:02:02.820669 Buzzy[4988:1717727] App startup timed out, reverting to last known good version
2016-11-13 21:02:02.841840 Buzzy[4988:1717079] Serving asset bundle version: 06ea28124eb8221400a40960079115860933896c
2016-11-13 21:02:04.037959 Buzzy[4988:1717601] Push Plugin register called
2016-11-13 21:02:04.038024 Buzzy[4988:1717601] PushPlugin.register: setting badge to false
2016-11-13 21:02:04.038044 Buzzy[4988:1717601] PushPlugin.register: clear badge is set to 0
2016-11-13 21:02:04.038058 Buzzy[4988:1717601] PushPlugin.register: better button setup
2016-11-13 21:02:04.038216 Buzzy[4988:1717601] GCM Sender ID (null)
2016-11-13 21:02:04.038241 Buzzy[4988:1717601] Using APNS Notification
2016-11-13 21:02:04.047700 Buzzy[4988:1717079] Push Plugin register success: <0d25f167 838005a8 eb1ccede df656199 a09f2400 42c890d9 07cf30c5 ade0288c>
2016-11-13 21:04:02.841952 Buzzy[4988:1718009] App startup timed out, reverting to last known good version
2016-11-13 21:04:02.857849 Buzzy[4988:1717079] Serving asset bundle version: 258c426bb6147c4fe43935834585825c1c5ef424
2016-11-13 21:04:04.470070 Buzzy[4988:1718029] Push Plugin register called
2016-11-13 21:04:04.470130 Buzzy[4988:1718029] PushPlugin.register: setting badge to false
2016-11-13 21:04:04.470151 Buzzy[4988:1718029] PushPlugin.register: clear badge is set to 0
2016-11-13 21:04:04.470165 Buzzy[4988:1718029] PushPlugin.register: better button setup
2016-11-13 21:04:04.470243 Buzzy[4988:1718029] GCM Sender ID (null)
2016-11-13 21:04:04.470260 Buzzy[4988:1718029] Using APNS Notification
2016-11-13 21:04:04.479448 Buzzy[4988:1717079] Push Plugin register success: <0d25f167 838005a8 eb1ccede df656199 a09f2400 42c890d9 07cf30c5 ade0288c>
2016-11-13 21:04:06.822769 Buzzy[4988:1717079] Start downloading asset manifest from: manifest.json -- https://buzzy.buzz/__cordova/
2016-11-13 21:04:08.013679 Buzzy[4988:1718009] Downloaded asset manifest for version: 06ea28124eb8221400a40960079115860933896c
2016-11-13 21:04:08.015318 Buzzy[4988:1718009] Download failure: Skipping downloading blacklisted version
2016-11-13 21:04:13.859149 Buzzy[4988:1717079] THREAD WARNING: ['Geolocation'] took '12.846924' ms. Plugin should use a background thread.
2016-11-13 21:06:02.858331 Buzzy[4988:1718694] App startup timed out, reverting to last known good version

```

**Comment by adamgins** [comment]
@martijnwalraven do you think it could be my timeout settings... 

```
App.setPreference("LoadUrlTimeoutValue", 120000);
App.setPreference("WebAppStartupTimeout", 120000);
```
Ie as I have it set to 2 mins... that's when folk are writing a response and then it's getting a blacklisted version and doing a refresh (or going to last known good version)?

I had originally increased the timeouts to remove the "white screen of death" issue. So a bit stuck on  a) why the downloads are getting blacklisted and b) what should I use as a timeout, so that I don't hit this issue. 

I also see alternate reloaders https://github.com/jamielob/reloader/ - so not sure if a config here with  manual refresh could solve the problem?

**Comment by matdurand** [comment]
This may be related to my issue #8063

**Comment by adamgins** [comment]
@matdurand thanks 
I have started using https://github.com/jamielob/reloader to try and workaround the issue.. ie let user choose when refresh with these params:


```
Reloader.configure({
    check: 'firstStart', // Check for new code every time the app starts
    checkTimer: 5000,  // Wait 3 seconds to see if new code is available
    refresh: 'start', 
    idleCutoff: 1000 * 60 * 10  // Wait 10 minutes before treating a resume as a start
});
```

seems to be working, most of the time https://github.com/jamielob/reloader/issues/12

**Comment by adamgins** [comment]
@martijnwalraven @benjamn Phew this issue is killing me...

I am using `Meteor 1.4.2.3` and I am still getting this issue.

I just refreshed my app store version and server versions. eg My server has 2.3.3.1 and app store has 2.3.2 ... however when I open the app it refreshes.. goes to 2.3.3.1 but then reverts to 2.3.2 

Any ideas pls?

**Comment by martijnwalraven** [comment]
@adamgins: Are you still getting `App startup timed out, reverting to last known good version` entries with the new server version? The way this timeout works is that it waits until all `Meteor.startup()` callbacks have been invoked. Once a version has timed out it is blacklisted and will not be downloaded again.

**Comment by adamgins** [comment]
@martijnwalraven thanks for the response. I think I may have found the issue. I had this code on the client:

```
if (Meteor.isCordova){
    Meteor.startup(function() {
        document.on('resume', function () {
            console.log("resume reconnect")
            Meteor.reconnect();
        });
    });

}
```

The goal was to and resolve reconnect issues I was facing. The issue is it seems like this code was breaking the `load-on-resume` sometimes. It seemed like to was dependent on the specific phone. (although not 100% sure) perhaps load/processing time for Cordova and `document` was not defined. This caused a JS error... and I think it was midstream reload-on-resume checking if there was a version... and seemed to constantly cause reloads. 

This is just theory / guess right now as it has been really hard to debug on users phones.

Not sure if you have any ideas if this is the case/possible?

One of the challenges was actually starting the Safari console to see the error.. as if/when it was replicated the reload would wipe it.. so I did not actually see the error until today.

Anyway, I am hoping this resolves the issue.  That said, not sure if anything can be done to catch this issue?

**Comment by rjakobsson** [comment]
Did that solve your issue @adamgins ?
I'm facing the same issue, I think.

**Comment by adamgins** [comment]
@rjakobsson when I was on the beta and had installed the plugin manually, it seemed to fix this... after moving off the beta and letting it install it, the problem has happened again.  Now that said, there should be no difference between the two scenarios, so perhaps it was never fixed in the first place. 

BTW, I am planning to move away from Cordova, too many issues...I am currently building a React Native app with ... a lot more effort and it's early days, but the initial React Native app seems a lot smoother, faster ... see https://github.com/inProgress-team/react-native-meteor this is now referenced in the guide too

**Comment by hwillson** [comment]
There hasn't been any activity on this issue in a while, and it's a bit difficult to pinpoint where the issue lies without a reproduction. As such, I'll close this issue for now. If anyone is still encountering this issue (with a recent version of Meteor) and can supply a reproduction demonstrating the problem, I'll happily re-open this issue. Thanks!
